design/registry-instead-of-s3.md

```diff
@@ -0,0 +1,50 @@
+# Optionally replace object store requirement with local registry
+
+KOTS requires an object store in order to store persistent versions of application version artifacts.
+KOTS also supports optionally providing a local OCI/Docker image repository to store application container images.
+This proposal removes the object store requirement in favor of the image repository.
+
+## Goals
+
+We could remove the requirement to have an object store (currently minio or rook) by using an image repository.
+This would be easier to support because there would be less third party components being used.
+
+## Non Goals
+
+
+## Background
+
+If an image repository is provided for the application, KOTS will create a data image containing the application version manifests and store it in the remote image repository.
+For online installs to existing clusters, if a registry is not provided, KOTS will deploy a small, PVC backed image repository.
+KOTS will no longer deploy minio or create the object store in Rook by default.
+
+## High-Level Design
+
+When an image repository is set in KOTS, the application manifests will be pushed to the registry instead of the object store.
+The code that downloads a version will be able to download a version from the image repository, instead.
+
+## Detailed Design
+
+New CLI flags will be added to `kots install` and `kots pull` to provide a local registry to be used.
+When these flags are not provided, KOTS will install Docker Distribution statefulset and service using a ClusterIP.
+This will be an insecure registry.
+
+
+Application manifests are currently packaged as a `.tar.gz` file.
+KOTS will continue to do this, but use the [ORAS Go Module](https://github.com/deislabs/oras#oras-go-module) to package these into an image.
```

Comment:
How about adding each file as a layer?  This opens possibility for server side deduplication.

Reply:
Hmm, that feels like a small optimization for a lot of complexity. A big application could have 1000s of files. With most YAML documenting being small, would there be a big win in deduplication?

An interesting alternative is to consider 1 image, where we just add each tgz to it, sequenced. So we could pull, add the sequence as a layer, and then push.

Reply:
yeah, ok.  good point.  images do have a limit on the number of layers.